### ISSUE TITLE
Assertion error on TextContainer 

### ISSUE THREAD

**Issue by afxgroup** [gemrb/gemrb#991]
I'm trying BG1 (on AmigaOS4) to test the engine and now when the game starts (after account creation) i have an assertion in TextContainer.cpp, line 223 printSize.w.
I've seen that printSize is created using  **Size printSize = layoutFont->StringSize(substr, &metrics);**
It could be a font problem? or a text problem?

The BG1 version is the Italian one.

**Comment by bradallred** [comment]
It's impossible to say without a copy of the font, the string, your encoding settings, and the `TextArea` in question.

you can try applying this fix from the other branch to see if that helps: 70eda9c

**Comment by afxgroup** [comment]
Nope. Same error. Where i can see which font is using and which text is trying to print? Is there a way to debug?

**Comment by bradallred** [comment]
use gdb or whatever other debugger. the fonts are all listed in the fonts.2da file. You can turn on `DEBUG_TEXT` in TextContainer.cpp and comment out the assert to facilitate debugging.

**Comment by lynxlynxlynx** [comment]
Did you manage to find at least which string/text is causing the crash? From that the font can be inferred, unless you specified one manually.

**Comment by lynxlynxlynx** [comment]
So does it now always crash when you save the game? By "click accept" you mean in the window asking you to name the save game (or confirm overwriting it?)?

**Comment by afxgroup** [comment]
Yes. Exactly

**Comment by bradallred** [comment]
> maybe is a known side effect

It's precisely the point of the debug flags for the most part and what I already mentioned above

Can you find the culprit string in your debugger and then enable only `DEBUG_TEXT` (disable the font one if its on),  comment out the assert, then screen shot the result so we can guess at what is likely happening?

**Comment by afxgroup** [comment]
Unlucky i don't have any debugger can i use except the ancient printf (Log) method and/or using -g flags plus addr2line command (that one i'm using). AmigaOS at moment doesn't have any debugger can help. I'll try to comment the assertions and see what happens. However why the "Prologue" is show only with debug_text enabled? and if i disable it is not shown?

**Comment by bradallred** [comment]
well, there is no reason you have to do this on Amiga. You will hit the same assert on Linux or Mac or whatever. The key to reproducing is just using the same version of the game with the same localization.

as to the "Prologue" thing. it 100% has nothing to do with `DEBUG_TEXT` look at the code yourself. `DEBUG_TEXT` just draws rectangles, nothing more.

**Comment by lynxlynxlynx** [comment]
@afxgroup do you have time to check with the latest code? IIRC you also had a bunch of amiga patches that we could now more easily integrate for posterity (platforms/ subdir).

**Comment by afxgroup** [comment]
I have to review everything since our C library has added support for wchar and a lot of other stuff. I'll keep you informed

**Comment by lynxlynxlynx** [comment]
magicsn has been updating the amiga port and successfully ran gemrb on it/them (modified 0.8.7, now working on more recent releases). The code is not upstream yet, but I doubt this specific issue will still be relevant, so closing.
